### DISCUSSION TITLE
Getting network nodes

### DISCUSSION THREAD

**Discussion by lynshi** [microsoft/CCF#4781] [Q&A] [answered]
[`/node/network/nodes`](https://github.com/microsoft/CCF/blob/4235d07e62e878f0943526cfd11bcda7ac915777/doc/schemas/node_openapi.json#L1197) allows the `host` and `port` filters. My recollection is that this is matched against the `published_address` of each node, not the `bind_address`.

However, there are also multiple RPC interfaces now, beyond just multiple "addresses" listed per node. If nodes have multiple RPC interfaces, is every `published_address` matched against the filter? If so, is the list of returned nodes deduplicated?

**Comment by achamayou** [comment, marked as answer]
@lynshi from a quick look at the logic, all published_interfaces are matched (https://github.com/microsoft/CCF/blob/main/src/node/rpc/node_frontend.h#L925), and there should not be any duplication, since there is a single foreach pass.

To be perfectly honest, I am not convinced of the value of this functionality, because:

1. This will only query nodes that are in the current KV. Now that node state cleanup is implemented, that should never be more than a few entries.
2. There are many good ways to filter JSON data, on the CLI (jq) or in libraries on the client side. Doing this on the server side is only marginally more convenient, and introduces extra TCB and user input parsing.

What do you think?

**Reply by lynshi** [reply]
Thanks for the clarification!

It seems to make sense to remove it but it will take some time for us to do the cleanup. We can accelerate the discussion for removing it if it affects API review, otherwise it is best to keep the filters for now.
